src/transformers/trainer.py

```diff
@@ -1819,7 +1820,7 @@ def _inner_training_loop(
                             self.scaler.step(self.optimizer)
                             self.scaler.update()
                         else:
-                            xm.optimizer_step(self.optimizer)
+                            self.optimizer.step()
```

Comment:
This change should not be there.

Reply:
As I explained in the previous comment, this is to decouple `xm.all_reduce` and `optimizer.step` in favor of the extra gradient clipping step in between.